samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -125,27 +123,16 @@ private void validateConfig() {
     // Add tables
     tables.forEach(spec -> jobGraph.addTable(spec, node));
 
-    jobGraph.validate();
+    if (!ApplicationUtil.isLegacyTaskApplication(appDesc)) {
+      // skip the validation when input streamIds are empty. This is only possible for LegacyApplication
+      jobGraph.validate();
+    }
 
     return jobGraph;
   }
 
   /**
-   * Figure out the number of partitions of all streams
-   */
-  /* package private */ void calculatePartitions(JobGraph jobGraph) {
-    // calculate the partitions for the input streams of join operators
-    calculateJoinInputPartitions(jobGraph, streamConfig);
-
-    // calculate the partitions for the rest of intermediate streams
-    calculateIntermediateStreamPartitions(jobGraph, config);
-
-    // validate all the partitions are assigned
-    validateIntermediateStreamPartitions(jobGraph);
-  }
-
-  /**
-   * Fetch the partitions of input/output streams and update the corresponding StreamEdges.
+   * Fetch the partitions of source/sink streams and update the StreamEdges.
```

Comment:
Maybe s/fetchInputAndOutputStreamPartitions/setInputAndOutputStreamPartitionCount for this method and javadoc to reflect that this method doesn't just fetch the count, it also sets it on the StreamSpec.

Reply:
Make sense. Thanks!